### PR TITLE
[LPT] nGraph nodes naming fix

### PR DESCRIPTION
**Pull request by eshoguli**
1. Naming fix
2. functional tests: FakeQuantize with dynamic intervals
3. functional tests: used kernel verification

### CODE REVIEW

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <ngraph/pass/visualize_tree.hpp>
+#include <transformations/low_precision/fake_quantize.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:15]: -> `<low_precision/fake_quantize.hpp>`

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 2]
```diff
@@ -1,9 +1,10 @@
-// Copyright (C) 2019-2020 Intel Corporation
+﻿// Copyright (C) 2019-2020 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include <transformations/op_conversions/convert_batch_to_space.hpp>
 #include <transformations/op_conversions/convert_space_to_batch.hpp>
+#include <regex>
```
Comment (eshoguli) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:7]: include order:
#include "layer_test_utils.hpp"
#include <regex>
#include <transformations/op_conversions/convert_batch_to_space.hpp>
#include <transformations/op_conversions/convert_space_to_batch.hpp>

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 3]
```diff
@@ -209,6 +210,34 @@ void LayerTestsCommon::Validate() {
     Compare(expectedOutputs, actualOutputs);
 }
 
+void LayerTestsCommon::checkKernel(std::string layerType, std::string permissibleKernel) {
```
Comment (eshoguli) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:213]: expectedKernel

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 4]
```diff
@@ -209,6 +210,34 @@ void LayerTestsCommon::Validate() {
...
+    const std::regex expectedKernel(permissibleKernel);
+    for (auto& targetOperationName : opNames) {
+        auto targetOperationKernel = getKernel(targetOperationName);
+        EXPECT_TRUE(std::regex_search(targetOperationKernel.begin(), targetOperationKernel.end(), expectedKernel))
+            << "Unexpected kernel. Actual: \"" << targetOperationKernel << "\" Expected: \"" << permissibleKernel << "\""
+            << " (Layer type: " << layerType << ", Layer name: " << targetOperationName << ")";
+    }
+}
+
+std::string LayerTestsCommon::getKernel(std::string layerName) {
```
Comment (eshoguli) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:233]: `const std::string& layerName`

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 5]
```diff
@@ -209,6 +210,34 @@ void LayerTestsCommon::Validate() {
...
+    auto perfomanceCounters = inferRequest.GetPerformanceCounts();
+    auto it = perfomanceCounters.find(layerName);
```
Comment (eshoguli) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:235]: `ASSERT_NE(perfomanceCounters.end(), it) << " performance counter for node " << layerName << " was not found";` (but not possible if method returns value) or `if (it == perfomanceCounters.end()) { THROW_IE_EXCEPTION << "..." }` (it's your case)

**Review by eshoguli — COMMENTED** [review]
Nothing urgent but find time to fix comments. Additional note: performance counter collection should be optional and NOT enabled by default.

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp]: Done, thanks!

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+const std::vector<ngraph::element::Type> precisions = {
+    ngraph::element::f32,
+    ngraph::element::i32,
+    ngraph::element::f16
+};
+
+const std::vector<FakeQuantizeWithDynamicIntervalsTransformationTestValues> fakeQuantizeTransformationTestValues = {
+    { LayerTransformation::createParamsU8I8(), false, false, false, true },
+    { LayerTransformation::createParamsU8I8(), true, false, false, false }
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:152]: Don't we need a case where all inputs are dynamic which is most common I suppose?

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        ngraph::Shape inputShape,
+        const bool inputLowConst,
+        const bool inpuHighConst,
+        const bool outputLowConst,
+        const bool outputHighConst) {
+        const auto input = std::make_shared<ngraph::opset1::Parameter>(precision, inputShape);
+        input->set_friendly_name("input");
+
+        const auto constantPresition = element::f32;
+        const auto constantShape = Shape{ 1, 1, 1, 1 };
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:108]: Are you sure that other Shapes will work correctly? What about per-channel quantization?

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        FakeQuantizeWithDynamicIntervalsTransformationTestValues testValues;
+        std::tie(precision, shape, testValues) = obj.param;
+
+        std::ostringstream result;
+        result << LayerTransformation::getTestCaseNameByParams(precision, shape, testValues.params) << testValues;
+        return result.str();
+    }
+
+private:
+    std::shared_ptr<ngraph::Node> makeTranspose(const ngraph::element::Type precision, const ngraph::Shape shape, const std::vector<float> low) {
```
Comment (dmitry-gorokhov) [inline inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:86]: Looks like this is common stuff and should be moved to ngraph builders.

File: inference-engine/src/low_precision_transformations/src/common/mat_mul.cpp [hunk 9]
```diff
@@ -95,7 +95,7 @@ bool MatMulTransformation::transform(TransformationContext &context, ngraph::pat
                 NetworkHelper::toScalar(as_type_ptr<opset1::Constant>(const1)),
                 const2)));
     replace_node(matMul, newMultiply);
-
+    newMultiply->set_friendly_name(matMul->get_friendly_name() + "/DequantizationMultiply");
```
Comment (dmitry-gorokhov) [inline inference-engine/src/low_precision_transformations/src/common/mat_mul.cpp:98]: Could you please explain the logic in which cases we need to update the name.  Why don't you do the same inside `add` transformation for example?

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp (hunk 8 above)
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:86]: Motivations to have private method: method size is small (3 lines), simple and not configurable. There are custom predefined values (precision, shape, transpose constant value). If move to common part then input arguments list will be huge.

File: inference-engine/src/low_precision_transformations/src/common/mat_mul.cpp (hunk 9 above)
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/src/common/mat_mul.cpp]: was removed, not the goal of this commit

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp (hunk 7 above)
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp:108]: It doesn't matter for this test. The tested source code doesn't use constant shape.

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp (hunk 6 above)
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp]: thanks, was added

File: inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp (hunk 8 above)
Comment (eshoguli) [reply inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp]: refactored, input node is used

File: inference-engine/src/low_precision_transformations/include/low_precision/network_helper.hpp [hunk 10]
```diff
@@ -160,6 +160,8 @@ class TRANSFORMATIONS_API NetworkHelper {
     // handles only specific case: Constant -> [dequantization operations] -> [node]
     static void foldDequantization(std::shared_ptr<Node>& node, const size_t branchIndex, const bool inPlace = false);
 
+    static std::shared_ptr<Node> mark_as_dequantization_op(std::shared_ptr<Node> op);
```
[the remark this answers is not in the document]
Comment (eshoguli) [reply inference-engine/src/low_precision_transformations/include/low_precision/network_helper.hpp]: thanks, done

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**